kadai1/imgconv/imgconv.go

```diff
@@ -0,0 +1,102 @@
+package imgconv
+
+import (
+	"fmt"
+	"image"
+	"image/jpeg"
+	"image/png"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+type Converter struct {
+	SrcExt string
+	DstExt string
+}
+
+var (
+	SupportExt = map[string]string{
+		"png":  ".png",
+		"jpg":  ".jpg",
+		"jpeg": ".jpg",
+	}
+	UnkownDecodeError = fmt.Errorf("Unkown Decode Error")
+	UnkownEncodeError = fmt.Errorf("Unkown Encode Error")
+)
+
+func ValidExt(ext string) bool {
+	_, ok := SupportExt[ext]
+	return ok
+}
+
+func NewConverter(srcExt, dstExt string) *Converter {
+	return &Converter{
+		SrcExt: SupportExt[srcExt],
+		DstExt: SupportExt[dstExt],
+	}
+}
+
+func (c *Converter) Convert(src string) error {
+	// 入力ファイルを取得する
+	srcfile, err := os.Open(src)
+	if err != nil {
+		return err
+	}
+	defer srcfile.Close()
+
+	// 読み出す(decode)
+	img, err := c.Decode(srcfile)
+	if err != nil {
+		return err
+	}
+
+	// 出力ファイルを作成する
+	dst := c.createDstFileName(src)
+	dstfile, err := os.Create(dst)
+	if err != nil {
+		return err
+	}
+	defer dstfile.Close()
+
+	// 書き出す(encode)
+	err = c.Encode(dstfile, img)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *Converter) Decode(srcfile *os.File) (image.Image, error) {
```

Comment:
`jpeg.Decode`と`png.Decode`が`io.Writer`を受け取るので、ここも`io.Writer`でいけそうです。(`*os.File`を受け取る仕様だと利用者の考えることが増えるので)
```suggestion
func (c *Converter) Decode(srcfile io.Writer) (image.Image, error) {
```

Reply:
よくみたらこれは課題2の範囲ですね🙇‍♂️

Reply:
いえ、ご指摘の通り、io.Writerの方が意図が伝わりやすそうですので、
こちら修正します！